src/server.js

```diff
@@ -23,12 +30,103 @@ const sendUserError = (err, res) => {
   }
 };
 
+const middleVerification = (req, res, next) => {
+  const sess = req.session;
+  if (!sess.LoginId) {
+    sendUserError("Please log in", res);
+    return;
+  }
+  User.findById(sess.LoginId, (err, user) => {
+    if (err) {
+      sendUserError(err, res);
+      return;
+    }
+    req.user = user;
+    next();
+  });
+};
+
+server.use("/restricted/*", (req, res, next) => {
+  const sess = req.session;
+  if (!sess.LoginId) {
+    sendUserError("Please log in", res);
+    return;
+  }
+  User.findById(sess.LoginId, (err, user) => {
+    if (err) {
+      sendUserError(err, res);
+      return;
+    }
+    req.user = user;
+  });
+});
+
 // TODO: implement routes
+server.post("/users", (req, res) => {
+  // getting the username and password from a user
+  const { username, password } = req.body;
+  // Check if the password is empty
+  if (password === "") {
```

Comment:
Here you can just do `if (!password)`, which will check for empty strings among other non-truthy values.